src/CallChain.js

```diff
@@ -0,0 +1,41 @@
+import chalk from "chalk"
+
+class CallChain {
+  constructor(name1, parent) {
+    this.name = name1
+    this.parent = parent
+  }
+
+  static create(name) {
+    return new CallChain(name)
+  }
+
+  add(name) {
+    return new CallChain(name, this)
+  }
+
+  hasParentName(name) {
+    return this.name === name || (this.parent && this.parent.hasParentName(name))
+  }
+
+  hasCyclic() {
+    const parent = this.parent
```

Comment:
Not really. Property access can be expensive depending on the property, so I've adopted the practice, but I doubt it makes a difference in this case.